pkg/distributor/distributor_test.go

```diff
@@ -4757,6 +4759,64 @@ func TestSeriesAreShardedToCorrectIngesters(t *testing.T) {
 	assert.Equal(t, series, totalMetadata) // each series has unique metric name, and each metric name gets metadata
 }
 
+func TestHandleIngesterPushError(t *testing.T) {
+	testErrorMsg := "this is a test error message"
+	outputErrorMsgPrefix := "failed pushing to ingester"
+	userID := "test"
+	errWithUserID := fmt.Errorf("user=%s: %s", userID, testErrorMsg)
+	test := map[string]struct {
+		ingesterPushError   error
+		expectedOutputError error
+	}{
+		"no error gives no error": {
+			ingesterPushError:   nil,
+			expectedOutputError: nil,
+		},
+		"an http 400 error gives an http 400 error": {
+			ingesterPushError:   httpgrpc.Errorf(http.StatusBadRequest, testErrorMsg),
+			expectedOutputError: httpgrpc.Errorf(http.StatusBadRequest, "%s: %s", outputErrorMsgPrefix, testErrorMsg),
+		},
+		"an http 500 error gives an http 500 error": {
+			ingesterPushError:   httpgrpc.Errorf(http.StatusInternalServerError, testErrorMsg),
+			expectedOutputError: httpgrpc.Errorf(http.StatusInternalServerError, "%s: %s", outputErrorMsgPrefix, testErrorMsg),
+		},
+		"grpc errMaxInflightRequestsReached error gives a wrapped errMaxInflightRequestsReached error": {
+			ingesterPushError:   errMaxInflightRequestsReached,
+			expectedOutputError: errors.Wrap(errMaxInflightRequestsReached, outputErrorMsgPrefix),
+		},
+		"grpc errMaxIngestionRateReached error gives a wrapped errMaxIngestionRateReached error": {
+			ingesterPushError:   errMaxIngestionRateReached,
+			expectedOutputError: errors.Wrap(errMaxIngestionRateReached, outputErrorMsgPrefix),
+		},
```

Comment:
This is a mistake, I accidentally used `errMaxIngestionRateReached` instead of `errMaxInflightRequestsReached`, which is returned by `Ingester.PushWithCleanup`. Basically my idea here was to show how all possible errors returned by `Ingester.PushWithCleanup()` would be handled by `Distributor.handleIngesterPushError()`

Reply:
Distributor will also not see those errors from ingester package directly. They always get translated by grpc layer.

Reply:
Implementation of `Ingester.PushWithCleanup()` was changed after the PR had been created. I will remove these test cases.